### ISSUE TITLE
validation dataset not working with R package

### ISSUE THREAD

**Issue by harpomaxx** [catboost/catboost#1406]
Problem: When using the R package of catboost  I cant use a validation dataset.
catboost version: 0.24.1
Operating System: Linux Ubuntu 18.04
CPU: Ryzen 2600

When I tried something like this example:

```
library(catboost)
data(iris)
train_idx<-sample(1:nrow(iris),(nrow(iris) * 70) / 100.0)
train<-iris[train_idx,]
val<-iris[-train_idx,]
pool_train = catboost.load_pool( train[,1:4], 
                                 label = as.numeric(as.factor(train$Species))-1)
pool_val = catboost.load_pool( val[,1:4], 
                               label = as.numeric(as.factor(val$Species))-1)
fit_params <- list(iterations = 100,
                   loss_function = 'MultiClass',
                   border_count = 32,
                   depth = 3,
                   learning_rate = 0.03,
                   l2_leaf_reg = 3.5,
                   task_type = 'CPU',
                   train_pool=pool_val,
                   verbose = 10)
model <- catboost.train(pool_train, params = fit_params)
```
I get an error:
`
Error: No method asJSON S3 class: catboost.Pool
`
If I remove the `train_pool` parameter, everything works.
Not sure if I should put the eval dataset in the params list. But according to documentation, it is the correct way to do it.
Regards,

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
According to https://github.com/catboost/catboost/blob/37533b8943060f673dd768891669e0a80fa51111/catboost/R-package/R/catboost.R#L393 , 'train_pool' is not listed as parameter that may be passed to catboost.train via params.

Could you please double check?

**Comment by harpomaxx** [comment]
Thanks for your help.

Apparently, the documentation from the website is outdated. The reference to `train_pool` in the `params` list is still there but it was removed from the code. Anyway, the correct way of passing the validation set is through the `test_pool` parameters. The name confused me, I thought it was the "actual test set" and not the "validation set". 

A working example 
```
library(catboost)
data(iris)
train_idx<-sample(1:nrow(iris),(nrow(iris) * 70) / 100.0)
train<-iris[train_idx,]
val<-iris[-train_idx,]
pool_train = catboost.load_pool( train[,1:4], 
                                 label = as.numeric(as.factor(train$Species))-1)
pool_val = catboost.load_pool( val[,1:4], 
                               label = as.numeric(as.factor(val$Species))-1)
fit_params <- list(iterations = 100,
                   loss_function = 'MultiClass',
                   border_count = 32,
                   depth = 3,
                   learning_rate = 0.03,
                   l2_leaf_reg = 3.5,
                   task_type = 'CPU',
                   verbose = 10)
model <- catboost.train(pool_train, pool_val, params = fit_params)
```
